packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -287,8 +288,15 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     options.envResources,
     options.sdk,
     options.overrideTemplate);
+  let bootstrapStackName: string | undefined;
+  try {
+    bootstrapStackName = (await options.envResources.lookupToolkit()).stackName;
```

Comment:
This performs a `lookupToolkit`, which returns ` ToolkitInfo`, which has most of the information already.

Why can't the logic live there?

Reply:
The answer was I tried but it's an abstract class and we have a "we don't exist yet" version of this class that throws errors. And when I put it in there a bunch of tests broke, so I just said screw it, and went back to my functional approach

Reply:
It's certainly not impossible but it was causing trouble. if you wanna say do it anyway I'll figure it out.